src/lib.rs

```diff
@@ -372,8 +418,8 @@ mod benches {
             let result =
                 select_coins_bnb(1003 * COIN, CHANGE_LOWER, &mut utxo_pool.clone()).unwrap();
             assert_eq!(2, result.len());
-            assert_eq!(1000 * COIN, result[0].value);
-            assert_eq!(3 * COIN, result[1].value);
+            assert_eq!(1000 * COIN, result[1].value);
+            assert_eq!(3 * COIN, result[0].value);
```

Comment:
Perhaps just check whether there is one coin of value 1'000 present, and one coin of value 3 present, rather than expecting them at specific positions.

This test made me realize that `combinations` deduplicates repeat values which is much better than my assumption above, but in the real world, UTXOs will often be similar in value but not match exactly. I'd be curious what sort of run time you would get in this bench if you had 30 UTXOs with distinct values instead.